crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -726,75 +742,160 @@ impl Inner {
             AlgebraicValue::String(seq_name.to_owned()),
         )
         .map(|mut iter| {
-            iter.next()
-                .map(|row| SequenceId(*row.view().elements[0].as_u32().unwrap()))
+            iter.next().map(|row| {
+                let id = row.view().elements[0].as_u32().unwrap();
+                (*id).into()
+            })
         })
     }
 
-    fn create_table(&mut self, table_schema: TableDef) -> super::Result<TableId> {
-        let table_name = table_schema.table_name.as_str();
-        log::trace!("TABLE CREATING: {table_name}");
+    fn create_constraint(&mut self, table_id: TableId, constraint: ConstraintDef) -> super::Result<ConstraintId> {
+        log::trace!(
+            "CONSTRAINT CREATING: {} for table: {} and cols: {:?}",
+            constraint.constraint_name,
+            table_id,
+            constraint.columns
+        );
 
-        if table_name_is_system(table_name) {
-            return Err(TableError::System(table_name.into()).into());
+        // Verify we have 1 column if need `auto_inc`
+        if constraint.constraints.has_autoinc() && constraint.columns.len() != 1 {
+            return Err(SequenceError::MultiColumnAutoInc(table_id, constraint.columns).into());
+        };
+
+        // Insert the constraint row into st_constraint
+        // NOTE: Because st_constraint has a unique index on constraint_name, this will
+        // fail if the table already exists.
+        let constraint_row = StConstraintRow {
+            constraint_id: 0.into(), // autogen'd
+            columns: constraint.columns.clone(),
+            constraint_name: constraint.constraint_name.clone(),
+            constraints: constraint.constraints,
+            table_id,
+        };
+
+        let row = ProductValue::from(constraint_row);
+        let result = self.insert(ST_CONSTRAINTS_ID, row)?;
+        let constraint_row = StConstraintRow::try_from(&result)?;
+        let constraint_id = constraint_row.constraint_id;
+
+        let mut constraint = ConstraintSchema::from_def(table_id, constraint);
+        constraint.constraint_id = constraint_id;
+
+        let insert_table = self.get_insert_table_mut(constraint.table_id)?;
+        insert_table.schema.update_constraint(constraint.clone());
```

Comment:
It could be called outside the normal table creation so, to avoid duplicates and get last results UPSERT